src/overlays/actors/ovl_Boss_Fd/z_boss_fd.c

```diff
@@ -23,275 +57,1937 @@ const ActorInit Boss_Fd_InitVars = {
+
+            for (i3 = 0; i3 < 3; i3++) {
+                Math_ApproachF(&this->rightArmRot[i3].x, spE0[i3].x, 1.0f, 1000.0f);
+                Math_ApproachF(&this->rightArmRot[i3].y, spE0[i3].y, 1.0f, 1000.0f);
+                Math_ApproachF(&this->leftArmRot[i3].x, spBC[i3].x, 1.0f, 1000.0f);
+                Math_ApproachF(&this->leftArmRot[i3].y, spBC[i3].y, 1.0f, 1000.0f);
+            }
+        } else {
+            for (i2 = 0; i2 < 3; i2++) {
+                phi_f20 = 0.0f;
+                Math_ApproachZeroF(&this->rightArmRot[i2].y, 0.1f, 100.0f);
+                Math_ApproachZeroF(&this->leftArmRot[i2].y, 0.1f, 100.0f);
+                if (i2 == 0) {
+                    phi_f20 = -3000.0f;
+                }
+                Math_ApproachF(&this->rightArmRot[i2].x, phi_f20, 0.1f, 100.0f);
+                Math_ApproachF(&this->leftArmRot[i2].x, -phi_f20, 0.1f, 100.0f);
+            }
+        }
+    }
+}
+#else
+static Vec3f D_808D19E0 = { 0.0f, 0.0f, 0.0f };
+static Vec3f D_808D19EC = { 0.0f, 0.03f, 0.0f };
+
+static Vec3f D_808D19F8 = { 0.0f, 0.0f, 0.0f };
+static Vec3f D_808D1A04 = { 0.0f, 0.0f, 0.0f };
+
+static Vec3f D_808D1A10 = { 0.0f, 0.0f, 0.0f };
+static Vec3f D_808D1A1C = { 0.0f, 0.03f, 0.0f };
+#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd/BossFd_Fly.s")
+#endif
+
+void BossFd_Wait(BossFd* this, GlobalContext* globalCtx) {
+    if (this->handoffSignal == FD2_SIGNAL_FLY) { // Set by BossFd2
+        u8 temp_rand;
+
+        this->handoffSignal = FD2_SIGNAL_NONE;
+        BossFd_SetupFly(this, globalCtx);
+        do {
+            temp_rand = Rand_ZeroFloat(8.9f);
+        } while (temp_rand == this->holeIndex);
+        this->holeIndex = temp_rand;
+        if (1) {} // Needed for matching
+        this->targetPosition.x = sHoleLocations[this->holeIndex].x;
+        this->targetPosition.y = sHoleLocations[this->holeIndex].y - 200.0f;
+        this->targetPosition.z = sHoleLocations[this->holeIndex].z;
+        this->actor.world.pos = this->targetPosition;
+
+        this->timers[0] = 10;
+        this->work[BFD_ACTION_STATE] = BOSSFD_EMERGE;
+        this->work[BFD_START_ATTACK] = true;
+    }
+    if (this->handoffSignal == FD2_SIGNAL_DEATH) {
+        this->handoffSignal = FD2_SIGNAL_NONE;
+        BossFd_SetupFly(this, globalCtx);
+        this->holeIndex = 1;
+        this->targetPosition.x = sHoleLocations[1].x;
+        this->targetPosition.y = sHoleLocations[1].y - 200.0f;
+        this->targetPosition.z = sHoleLocations[1].z;
+        this->actor.world.pos = this->targetPosition;
+        this->timers[0] = 10;
+        this->work[BFD_ACTION_STATE] = BOSSFD_EMERGE;
+    }
+}
+
+static Vec3f sFireAudioVec = { 0.0f, 0.0f, 50.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd/func_808CAF5C.s")
+void BossFd_Effects(BossFd* this, GlobalContext* globalCtx) {
+    static Color_RGBA8 colorYellow = { 255, 255, 0, 255 };
```

Comment:
is this updating the effects? might be good to name the function after that fact

Reply:
It's a catch-all function that handles spawning various effects, updating Volvagia's face, and changing the environment variables. Not quite sure how to name it.